File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    var answer = [];
    
    var numer = ''
    var denom = ''

    numer = numer1 * denom2 + numer2 * denom1
    denom = denom1 * denom2

    for (let i = denom; i > 0; i--) {
        if (numer % i == 0 && denom % i == 0) {
            numer = numer/i
            denom = denom/i
        }
    }
    
    return answer = [numer, denom]
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        
        answer[0] = (numer1 * denom2) + (numer2 * denom1);
        answer[1] = denom1 * denom2;
        
        int gcd = findGcd(answer[0], answer[1]);
        
        answer[0] = answer[0] / gcd;
        answer[1] = answer[1] / gcd;
        
        return answer;
    }
    
    private int findGcd(int x, int y) {
        if (x < y) {
            int temp = x;
            x = y;
            y = temp;
        }
        
        while (y != 0) {
            int r = x % y;
            x = y;
            y = r;
        }
        
        return x;
    }
    
}